spark-shell/src/main/scala/org/apache/mahout/sparkbindings/shell/MahoutSparkILoop.scala

```diff
@@ -48,55 +77,63 @@ class MahoutSparkILoop extends SparkILoop {
 
     conf.set("spark.executor.memory", "1g")
 
-    sparkContext = mahoutSparkContext(
+    _interp.sparkContext = mahoutSparkContext(
       masterUrl = master,
       appName = "Mahout Spark Shell",
       customJars = jars,
       sparkConf = conf
     )
 
-    echo("Created spark context..")
+    echoToShell("Created spark context..")
     sparkContext
   }
 
+  // need to change our SparkDistributedContext name to 'sc' since  we cannot override the
+  // private sparkCleanUp() method.
+  // this is technically not part of Sparks explicitly defined Developer API though
+  // nothing in the SparkILoopInit.scala file is marked as such.
   override def initializeSpark() {
-    intp.beQuietDuring {
-      command("""
+    _interp.beQuietDuring {
+      _interp.interpret("""
 
-         @transient implicit val sdc: org.apache.mahout.math.drm.DistributedContext =
+         @transient implicit val sc: org.apache.mahout.math.drm.DistributedContext =
```

Comment:
> comes to that, we could potentially initialize sc with spark context and sdc with mahout context. perhaps that would be ok and create compatibility throughout.

Are you thinking something like:

```
@transient val sc = {
           val _sc = org.apache.spark.repl.Main.interp.createSparkContext()
           println("Spark context available as sc.")
           _sc
         }
@transient implicit val sdc: org.apache.mahout.math.drm.DistributedContext =
            new org.apache.mahout.sparkbindings.SparkDistributedContext(sc)

        echoToShell("Mahout distributed context is available as \"implicit val sdc\".")
```

I'm not sure where we'd be able to call `sdc.close()`  but I'm not sure that that would be necessary anyway since that method in turn simply calls `sc.stop()` which is handled by `SparkILoop.sparkCleanUp()`